File: BOJ/1260_dfs와 bfs.cpp
#include<stdio.h>
#include<cstring>
#include<string>
#include<algorithm>
#include<vector>
#include<queue>
#include<iostream>
using namespace std;
int n, m, a, b,v;
int chk1[1001], chk2[1001];
queue <int> q;
vector <vector<int>> adj(1001);
void dfs(int now)
{
	chk1[now] = 1;
	printf("%d ", now);
	for (int i = 0; i<adj[now].size(); i++)
	{
		int next = adj[now][i];
		if (chk1[next] != 1)
			dfs(next);
	}
}
void bfs(int now)
{
	chk2[now] = 1;
	q.push(now);
	while (!q.empty())
	{
		int here = q.front();
		q.pop();
		printf("%d ", here);
		for (int i = 0; i < adj[here].size(); i++)
		{
			int next = adj[here][i];
			if (chk2[next] != 1)
			{
				chk2[next] = 1;
				q.push(next);
			}
		}
	}
}
int main()
{
	cin >> n >> m >> v;
	for (int i = 1; i <= m; i++)
	{
		cin >> a >> b;
		adj[a].push_back(b);
		adj[b].push_back(a);
	}
	for (int i = 1; i <= n; i++)
		sort(adj[i].begin(), adj[i].end());
	dfs(v);
	printf("\n");
	bfs(v);
	printf("\n");
	return 0;
}

File: BOJ/1260_DFS와 BFS.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.ArrayList;
import java.util.LinkedList;
import java.util.Queue;
import java.util.StringTokenizer;

public class Main {
    static StringBuilder sb = new StringBuilder();
    static ArrayList<Integer>[] arrayList;
    static boolean[] visited;
    static int N, M;
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringTokenizer st = new StringTokenizer(br.readLine(), " ");
        N = Integer.parseInt(st.nextToken());
        M = Integer.parseInt(st.nextToken());
        int startVertex = Integer.parseInt(st.nextToken());
        arrayList = new ArrayList[N+1];
        for(int i=1;i<=N;i++) {
            arrayList[i] = new ArrayList<>();
        }
        for(int i=0;i<M;i++) {
            StringTokenizer edge = new StringTokenizer(br.readLine(), " ");
            int vertex1 = Integer.parseInt(edge.nextToken());
            int vertex2 = Integer.parseInt(edge.nextToken());
            arrayList[vertex1].add(vertex2);
            arrayList[vertex2].add(vertex1);
        }
        for(int i=1;i<=N;i++) {
            arrayList[i].sort(Integer::compareTo);
        }
        visited = new boolean[N+1];
        dfs(startVertex);
        sb.append("\n");
        visited = new boolean[N+1];
        bfs(startVertex);
        System.out.println(sb);
    }

    static void dfs(int now) {
        visited[now] = true;
        sb.append(now).append(" ");
        for(int next : arrayList[now]) {
            if(!visited[next]) {
                visited[next] = true;
                dfs(next);
            }
        }
    }

    static void bfs(int startVertex) {
        Queue<Integer> queue = new LinkedList<>();
        queue.add(startVertex);
        visited[startVertex] = true;
        while(!queue.isEmpty()) {
            int now = queue.poll();
            sb.append(now).append(" ");
            for(int next : arrayList[now]) {
                if(!visited[next]) {
                    visited[next] = true;
                    queue.add(next);
                }
            }
        }
    }
}
